api/app/controllers/api/v1/ingests_controller.rb

```diff
@@ -0,0 +1,53 @@
+module API
+  module V1
+    # Trigger ingestion processing via the API
+    class IngestsController < ApplicationController
+      before_action :set_project, only: [:create]
+      before_action :set_text, only: [:create]
+
+      resourceful! Ingestion do
+        @project&.ingestions
+        @text&.ingestions
```

Comment:
See below for why one-size-fits-all is not an ideal approach, but this also won't necessarily do what you think: ruby has implicit return for the last statement in a lexical block, so if this were called for a project, the block would still return nil, because `@text&.ingestions` is the last statement, and it'd be nil in that case.

If this were all to be the same controller, you'd want `@project&.ingestions || @text&.ingestions`.